torch/csrc/autograd/engine.cpp

```diff
@@ -345,14 +345,15 @@ static variable_list call_function(FunctionTask& task) {
   if(!task.base->keep_graph) {
     fn.will_release_variables();
   }
-  auto outputs = fn(inputs);
+  variable_list local_inputs = inputs;
+  auto outputs = fn(std::move(inputs));
```

Comment:
This has the downside that we now have an extra copy in the critical path, which wasn't there before. We should really avoid it when there are no callbacks registered (which is the usual case)